src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -87,16 +87,41 @@ export class IndexPattern implements IIndexPattern {
     title: ES_FIELD_TYPES.TEXT,
     timeFieldName: ES_FIELD_TYPES.KEYWORD,
     intervalName: ES_FIELD_TYPES.KEYWORD,
-    fields: 'json',
-    sourceFilters: 'json',
+    fields: {
```

Comment:
I'm not crazy about this code but will refactor in a future PR.

Reply:
what's this change about ? i don't understand the purpose.

if its not needed i would remove it, we should try to get to a toSpec() function which returns minimal serializable index pattern representation and specToIndexPattern function which takes that spec and gives you back an index pattern instance, everything else we should leave for follow up PRs. 

to do that we shouldn't need to change any internals of index patterns apart from adding those two functions

Reply:
This code is responsible for serializing and deserialzing the individual fields.